platform/viewer/src/lib/DICOMSR/retrieveDataFromSR.js

```diff
@@ -0,0 +1,70 @@
+import * as dcmjs from 'dcmjs';
+import { getAllDisplaySets, getInstanceMetadata } from './srUtils';
+
+const retrieveDataFromSR = Part10SRArrayBuffer => {
+  const allDisplaySets = getAllDisplaySets();
+
+  // Get the dicom data as an Object
+  const dicomData = dcmjs.data.DicomMessage.readFile(Part10SRArrayBuffer);
+  const dataset = dcmjs.data.DicomMetaDictionary.naturalizeDataset(
+    dicomData.dict
+  );
+
+  // Convert the SR into the kind of object the Measurements package is expecting
+  return imagingMeasurementsToMeasurementData(dataset, allDisplaySets);
+};
+
+const imagingMeasurementsToMeasurementData = (dataset, displaySets) => {
+  const { MeasurementReport } = dcmjs.adapters.Cornerstone;
+  const storedMeasurementByToolType = MeasurementReport.generateToolState(
+    dataset
+  );
```

Comment:
Oh, interesting. Do we do anything to check that data in the SR maps to something our application can display? (not that this would be the ideal place to do that, just thinking out loud.) For example, how should we handle:

There is Length data in the structured report, but...

- No Length `cornerstone-tool` is "added" and "enabled"
- OR no Length `vtk.js` widget
- OR `cornerstone-tool`'s Length tool expects a different tool state "shape"
    - We modify it at a later date, or the user has a custom Length tool

Reply:
hmmm 
The only thing we do right now is to check if dcmjs support certain tool when we are saving, as we use dcmjs to retrieve as well, the code believes all that is comming is supported as well, not sure how to handle different than that right now. 

> No Length cornerstone-tool is "added" and "enabled"
and
> OR no Length vtk.js widget
I think ifs the tool is not enabled/added, the toolState will be added but nothing wll happen, right? 

> cornerstone-tool's Length tool expects a different tool state "shape"
We also dont treat this scenario right now, I believe we would just add a "wrong" data into toolState